ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/MaterializedViewRewritingRelVisitor.java

```diff
@@ -50,7 +49,7 @@ public class MaterializedViewRewritingRelVisitor extends RelVisitor {
 
 
   private boolean containsAggregate;
-  private boolean fullAcidView;
+  private final boolean fullAcidView;
```

Comment:
others don't have to be final?

Reply:
This class is not stateless. Fields
```
containsAggregate
rewritingAllowed
countIndex
```
are set during plan traversal but `fullAcidView` should not be changed since it's value depends on the MV metadata.

Reply:
👍